src/main/java/Duke.java

```diff
@@ -1,10 +1,236 @@
+        f.mkdir();
+        new Duke("data/tasks.txt").run();
+    }
+    /**
+     * Executes the command and returns the result.
+     * @param s user command
+     * @return result of the command
+     */
+    public void todo(ArrayList<Task>list,String s) throws ErrorException{
+        if (s.length() == 4) {
+            throw new ErrorException("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+        else {
+            String td = s.substring(5);
+            Task t = new Todo(td);
+            list.add(t);
+            ui.drawline();
+            System.out.println("     Got it. I've added this task:");
+            System.out.println("     " + t);
+            System.out.println("     Now you have " + list.size() + " tasks in the list.");
+            ui.drawline();
+            storage.addtask(tasks.getlist());
+        }
+    }
+    /**
+     * Executes the command and returns the result.
+     * @param s user command
+     * @return result of the command
+     */
+    public void deadline(ArrayList<Task>list,String s) throws ErrorException{
+        if (s.length() == 8) {
+            throw new ErrorException("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+        else {
+            int sIndex = s.indexOf("/");
+            int firstIndex = sIndex - 1;
+            int secondIndex = sIndex + 4;
+            String deadlineDescription = s.substring(9, firstIndex);
+            String deadlineBy = s.substring(secondIndex);
+            Deadline d = new Deadline(deadlineDescription, deadlineBy);
+            tasks.timeform(deadlineBy,d);
+        }
+    }
+    /**
+     * Executes the command and returns the result.
+     * @param s user command
+     * @return result of the command
+     */
+    public void event(ArrayList<Task>list,String s) throws ErrorException{
+        if (s.length() == 5) {
+            throw new ErrorException("☹ OOPS!!! The description of a todo cannot be empty.");
+        }
+        else {
+            int sIndex = s.indexOf("/");
+            int firstIndex = sIndex - 1;
+            int secondIndex = sIndex + 4;
+            String eventDescription = s.substring(6, firstIndex);
+            String eventAt = s.substring(secondIndex);
+            Event event = new Event(eventDescription, eventAt);
+            tasks.timeform(eventAt,event);
+
+
+
+        }
+    }
+    /**
+     * Executes the command and returns the result.
+     * @param s user command
+     * @return result of the command
+     */
+    public void other(String s) throws ErrorException{
+        throw new ErrorException("☹ OOPS!!! I'm sorry, but I don't know what that means :-(");
+    }
+    /**
+     * Executes the command and returns the result.
+     * @param p user command
+     * @return result of the command
+     */
+    public void delete(ArrayList<Task>list,int p){
+        Task rt = list.get(p);
+        ui.drawline();
+        System.out.println("     Noted. I've removed this task:");
```

Comment:
Can also put this formatted printing as a function in the `UI` class, since you already have the class. Same for the rest